### PR TITLE
cratedb: init at 5.9.5

### LINKED ISSUE #116533: packaging request: crate db
**Project description**

> CrateDB is a distributed SQL database that makes it simple to store and analyze massive amounts of machine data in real-time.
>
> CrateDB offers the scalability and flexibility typically associated with a NoSQL database, is designed to run on inexpensive commodity servers and can be deployed and run on any sort of network - from personal computers to multi-region hybrid clouds.
>
> The smallest CrateDB clusters can easily ingest tens of thousands of records per second. The data can be queried, ad-hoc, in parallel across the whole cluster in real time.

**Metadata**

* main dependencies: Java >=11 and Python >=3.7
* build instrs: https://github.com/crate/crate/blob/master/devs/docs/basics.rst
* homepage URL: https://crate.io/
* source URL: on Github, https://github.com/crate/crate; tag name = `${version}`
* license: Apache-2.0
* platforms: `[x64 aarch64] && [linux windows mac]`

### PR DESCRIPTION
**Pull request by pluiedev**
Closes #116533 after over 3 years (!)

Should probably also add a service module

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cr/cratedb/package.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  postFixup = ''
+    # By default it wants to write to $out/data and $out/logs. Bad program.
+    wrapProgram $out/bin/crate \
+      --set CRATE_DISABLE_GC_LOGGING 1 \
+      --add-flags '-Cpath.data="''${XDG_DATA_HOME:-$HOME/.local/share}/crate"' \
+      --add-flags '-Cpath.logs="''${XDG_STATE_HOME:-$HOME/.local/state}/crate"'
+
+    wrapProgram $out/bin/crate-node \
+      --add-flags '-Cpath.data="''${XDG_DATA_HOME:-$HOME/.local/share}/crate"' \
+      --add-flags '-Cpath.logs="''${XDG_STATE_HOME:-$HOME/.local/state}/crate"'
```
Comment (getchoo) [inline pkgs/by-name/cr/cratedb/package.nix:60]:
```suggestion
    for bin in $out/bin/crate{,-node}; do
      wrapProgram $bin \
        --set CRATE_DISABLE_GC_LOGGING 1 \
        --add-flags '-Cpath.data="''${XDG_DATA_HOME:-$HOME/.local/share}/crate"' \
        --add-flags '-Cpath.logs="''${XDG_STATE_HOME:-$HOME/.local/state}/crate"'
    done
```
Nit

File: pkgs/by-name/cr/cratedb/package.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  version = "5.9.3";
+in
+maven.buildMavenPackage {
+  pname = "cratedb";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "crate";
+    repo = "crate";
+    rev = "refs/tags/${version}";
```
Comment (getchoo) [inline pkgs/by-name/cr/cratedb/package.nix:22]:
```suggestion
    tag = version;
```
Nit

File: pkgs/by-name/cr/cratedb/package.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      associated with NoSQL databases. Modest CrateDB clusters can ingest tens of thousands of records
+      per second without breaking a sweat. You can run ad-hoc queries using standard SQL.
+      CrateDB's blazing-fast distributed query execution engine parallelizes query workloads across
+      the whole cluster.
+    '';
+    homepage = "https://cratedb.com/database";
+    changelog = "https://cratedb.com/docs/crate/reference/en/latest/appendices/release-notes/${version}.html";
+    license = with lib.licenses; [ asl20 ];
+    maintainers = with lib.maintainers; [ pluiedev ];
+    inherit (jdk.meta) platforms;
```
Comment (getchoo) [inline pkgs/by-name/cr/cratedb/package.nix:79]:
```suggestion
```
The builder [sets this for us](https://github.com/NixOS/nixpkgs/blob/7c80d8a1c2ae0d7e6a2aab554aad87b36a8f7486/pkgs/by-name/ma/maven/build-maven-package.nix#L103)

File: pkgs/by-name/cr/cratedb/package.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    mainProgram = "crate";
```
Comment (getchoo) [inline pkgs/by-name/cr/cratedb/package.nix:80]:
```suggestion
    mainProgram = "crate";
    broken = stdenv.hostPlatform.isDarwin;
```
Even the maven deps fail on Darwin so...

**Review by getchoo — CHANGES_REQUESTED** [review]

File: pkgs/by-name/cr/cratedb/package.nix [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      CrateDB's blazing-fast distributed query execution engine parallelizes query workloads across
+      the whole cluster.
+    '';
+    homepage = "https://cratedb.com/database";
+    changelog = "https://cratedb.com/docs/crate/reference/en/latest/appendices/release-notes/${version}.html";
+    license = with lib.licenses; [ asl20 ];
+    maintainers = with lib.maintainers; [ pluiedev ];
+    mainProgram = "crate";
+    # Deps are broken on Darwin
+    broken = stdenv.hostPlatform.isDarwin;
```
Comment (getchoo) [inline pkgs/by-name/cr/cratedb/package.nix:78]:
```suggestion
    broken = stdenv.hostPlatform.system != "x86_64-linux";
```
Don't know how else to express this well. Seems the maven deps are architecture dependent :/

Comment (pluiedev) [reply pkgs/by-name/cr/cratedb/package.nix:78]: I think I'm just going to set the platform to `x86_64-linux` only for now

**Review by getchoo — APPROVED** [review]
Diff LGTM

**Comment by pluiedev** [comment]
Note: I'm aware that version 5.9.5 has been released, but I don't really want to sit for 40+ minutes just to get the new hashes. Perhaps @r-ryantm would be so kind as to do that for me one day

**Review by wolfgangwalther — COMMENTED** [review]
> Note: I'm aware that version 5.9.5 has been released, but I don't really want to sit for 40+ minutes just to get the new hashes. Perhaps @r-ryantm would be so kind as to do that for me one day

I am not r-ryantm, but I have hashes for you!

**Pull request merged**